### PR TITLE
[owlwatch] Support object types in mappings and limit the nested properties

### PR DESCRIPTION
**Pull request by acs**
@alpgarcia we need those changes to tests all panel JSON files that we have now. Probably with nested properties of more that one level, we should analyze them recursively, but I want to talk with you before investing more effort.

Please take a look to the changes and modify anything you want!

### CODE REVIEW

File: src/owlwatch/schema/model.py [hunk 1]
```diff
@@ -180,10 +180,20 @@ def from_json(cls, ip_json):
         # Get index pattern fields
         fields_json = json.loads(ip_json['value']['fields'])
         for json_field in fields_json:
+            if 'scripted' in json_field and json_field['scripted']:
+                # Don't check Kibana generated scripted fields
+                continue
+            elif 'is_' in json_field['name']:
```
Comment (alpgarcia) [inline src/owlwatch/schema/model.py]: This should be done via Schema.__excluded_props . Even better, using another exclusion list, passed as parameter to Schema class at some point, so the user would be able to exclude custom lists of fields.

File: src/owlwatch/schema/model.py [hunk 2]
```diff
@@ -180,10 +180,20 @@ def from_json(cls, ip_json):
...
+                # is_gmane only exists in mbox coming from gmane for example
+                continue
+            agg = True
+            if 'aggregatable' in json_field:
+                agg = json_field['aggregatable']
             index_pattern.add_property(pname=json_field['name'],
                                        ptype=json_field['type'],
                                        analyzed=json_field['analyzed'],
-                                       agg=json_field['aggregatable'])
+                                       agg=agg)
```
Comment (alpgarcia) [inline src/owlwatch/schema/model.py]: If an Index Pattern doesn't have aggregatable field, it is probably due to an old version of JSON file, not exported from Kibana 5. Thus, we should raise an error or warning and just not adding that property.

File: src/owlwatch/schema/model.py [hunk 3]
```diff
@@ -264,10 +274,16 @@ def from_json(cls, index_name, mapping_json):
                         prop_name = prop + '.' + nested_prop
                         analyzed = None
                         agg = None
+
                         if 'fielddata' in nested_value:
                             agg = nested_value['fielddata']
+                        if 'type' in nested_value:
+                            ptype = nested_value['type']
+                        else:
+                            print('Not adding to es_mapping value: %s' % nested_value)
```
Comment (alpgarcia) [inline src/owlwatch/schema/model.py]: As this is (intends to be) a library, we should use logging for writing messages. I'll also explain why we are not adding that property in the message.

**Comment by acs** [comment]
Modify all according to your comments and our meeting today. 

Could you review it @alpgarcia ? Thanks man!

**Comment by alpgarcia** [comment]
Thanks @acs !! LGTM!!

We probably need to add support for excluding fields from outside the model, in such a way that the caller knows what she is excluding. I'll open an issue to improve that exclusion thing :)

**Pull request merged**